TCAT/Controllers/RouteDetailDrawerViewController+Extensions.swift

```diff
@@ -101,43 +108,59 @@ extension RouteDetailDrawerViewController: UITableViewDataSource {
         }
-
-        switch directionsAndVisibleStops[indexPath.row] {
-        case .busStop(let busStop):
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.busStopDetailCellIdentifier) as? BusStopTableViewCell
-                else { return UITableViewCell ()}
-            cell.configure(for: busStop.name)
-            return format(cell)
-        case .direction(let direction):
-            if direction.type == .walk || direction.type == .arrive {
-                guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.smallDetailCellIdentifier, for: indexPath) as? SmallDetailTableViewCell
-                    else { return UITableViewCell() }
-                cell.configure(for: direction,
-                               isFirstStep: indexPath.row == 0,
-                               isLastStep: indexPath.row == directionsAndVisibleStops.count - 1)
-                return format(cell)
-            } else {
-                let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.largeDetailCellIdentifier) as! LargeDetailTableViewCell
-                cell.configure(
-                    for: direction,
-                    isFirstStep: indexPath.row == 0,
-                    isExpanded: expandedDirections.contains(direction),
-                    delegate: self
-                )
+        
+        switch sections[indexPath.section] {
+        case .routeDetail:
+            switch directionsAndVisibleStops[indexPath.row] {
+            case .busStop(let busStop):
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.busStopDetailCellIdentifier) as? BusStopTableViewCell
+                    else { return UITableViewCell()}
+                cell.configure(for: busStop.name)
                 return format(cell)
+            case .direction(let direction):
+                if direction.type == .walk || direction.type == .arrive {
+                    guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.smallDetailCellIdentifier, for: indexPath) as? SmallDetailTableViewCell
+                        else { return UITableViewCell() }
+                    cell.configure(for: direction,
+                                   isFirstStep: indexPath.row == 0,
+                                   isLastStep: indexPath.row == directionsAndVisibleStops.count - 1)
+                    return format(cell)
+                } else {
+                    let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.largeDetailCellIdentifier) as! LargeDetailTableViewCell
+                    cell.configure(
+                        for: direction,
+                        isFirstStep: indexPath.row == 0,
+                        isExpanded: expandedDirections.contains(direction),
+                        delegate: self
+                    )
+                    return format(cell)
```

Comment:
So I'm lookin at the format function that we have and all it does is set `cell.selectionStyle = .none`. Can we just move this line into the init of the cells?